label_studio/ml/api_connector.py

```diff
@@ -31,18 +31,37 @@
 TIMEOUT_DELETE = float(get_env('ML_TIMEOUT_DELETE', 1))
 TIMEOUT_TRAIN_JOB_STATUS = float(get_env('ML_TIMEOUT_TRAIN_JOB_STATUS', 1))
 
+# TODO
+# we would need to make it configurable on the ML backend side too
+PREDICT_URL = 'predict'
+HEALTH_URL = 'health'
+VALIDATE_URL = 'validate'
+SETUP_URL = 'setup'
+DUPLICATE_URL = 'duplicate_model'
+DELETE_URL = 'delete'
+JOB_STATUS_URL = 'job_status'
+VERSIONS_URL = 'versions'
+
 
 class BaseHTTPAPI(object):
     MAX_RETRIES = 2
     HEADERS = {
         'User-Agent': 'heartex/' + (version or ''),
     }
 
-    def __init__(self, url, timeout=None, connection_timeout=None, max_retries=None, headers=None, **kwargs):
+    def __init__(
+        self, url, timeout=None, connection_timeout=None, max_retries=None, headers=None, auth_method=None, **kwargs
+    ):
         self._url = url
         self._timeout = timeout or TIMEOUT_DEFAULT
         self._connection_timeout = connection_timeout or CONNECTION_TIMEOUT
         self._headers = headers or {}
+        self._auth_method = auth_method
+
+        if auth_method == 'BA':
```

Comment:
Here and everywhere in the PR: we shouldn't refer to strings from an enum explicitly. Instead, let's compare `auth_method` with `MLBackendAuth.BASIC_AUTH`

Ideally we would use an enum on the frontend too for the same reason.